src/radical/pilot/agent/scheduler/continuous.py

```diff
@@ -422,9 +423,16 @@ def schedule_task(self, task):
             #   - if the previous use included this node
             # If a tag exists, continue to consider this node if the tag was
             # used for this node - else continue to the next node.
-            if tag is not None and tag in self._tag_history:
-                if node_uid not in self._tag_history[tag]:
-                    continue
+            if tag is not None:
+                if tag in self._tag_history:
+                    if node_uid not in self._tag_history[tag]:
+                        continue
+                # for a new tag check that nodes were not used for previous tags
+                else:
+                    # `exclusive` -> not to share nodes between different tags
+                    is_exclusive = td.get('tags', {}).get('exclusive', True)
+                    if is_exclusive and node_uid in self._tagged_nodes:
+                        continue
```

Comment:
Should we maybe set the default to `False`?  If there are more tags than nodes, the execution will stall otherwise.  Maybe that's also a check we should perform (when we add new tags to the exclusive list):
```py
if len(self._tagged_nodes) >= len(self._nodes):
  task.state = rp.FAILED
  ...
```
although that would only work when all nodes in the `_tagged_nodes` are tagged for exclusivity - which we don't know.  Any idea?

Actually, what happens then - will `_iterate_nodes` cycle around and we are stuck in this loop forever?

Reply:
good points, thanks! didn't think about that... I agree to keep default value as `False` and will think more about exception case (first thought: if there are not more exclusive nodes for new tags, then let it go, but print a warning in logs)